consensus/parlia/parlia.go

```diff
@@ -1062,7 +1062,8 @@ func (p *Parlia) distributeIncoming(val common.Address, state *state.StateDB, he
 	coinbase := header.Coinbase
 	balance := state.GetBalance(consensus.SystemAddress)
 	if balance.Cmp(common.Big0) <= 0 {
-		return nil
+		// BEP-176, if reward = 0, just trigger reward block count
+		return p.distributeToValidator(common.Big0, val, state, header, chain, txs, receipts, receivedTxs, usedGas, mining)
```

Comment:
This is expected to be a hard fork logic.

Reply:
I believe you did not test this PR yet, because now  `p.distributeToValidator` happens before `state.AddBalance(coinbase, balance)`, which means there is no fund to do the deposit.

Reply:
yes, I miss the hard fork check, it only takes effect in the new hard fork blocks.